app/controllers/assignments_controller.rb

```diff
@@ -37,20 +34,18 @@ def start
   # GET /assignments/new
   def new
     @assignment = Assignment.new
+    @assignment.group_id = params[:group_id]
     @assignment.deadline = Time.now+1.week
+    authorize @assignment, :admin_access?
   end
 
   # GET /assignments/1/edit
   def edit
-    if @assignment.status == 'closed'
-      render plain: "access restricted" and return;
-    end
+    authorize @assignment
   end
 
   def reopen
-    if @assignment.status == 'open'
-      render plain: "already open" and return;
-    end
+    authorize @assignment
```

Comment:
Again, I might be wrong here, we are already checking for admin access so why do need to authorize again?

Reply:
See `assignment_policy` file, `reopen?` has another check to see if the assignment isn't already open and it raises its own custom error [here](https://github.com/CircuitVerse/CircuitVerse/pull/184/files#diff-c466b4128cdaa10a4f59eb1cb6be506cR30) That is why it is needed here again.

Reply:
Yes, I realized you have moved some of the non-authorization logic to the policies